python/tchannel/tornado/stream.py

```diff
@@ -0,0 +1,143 @@
+import tornado
+import tornado.gen
+import tornado.ioloop
+from tornado.iostream import PipeIOStream, StreamClosedError
+from ..exceptions import StreamingException
+from ..messages.common import StreamState
+from ..messages import common
+
+try:
+    from tornado.locks import Condition
+except ImportError:  # pragma: no cover
+    from toro import Condition
+
+
+class ArgStream(object):
+
+    def __init__(self):
+        self._state = StreamState.init
```

Comment:
Move `_state` initialization into child classes (even if that leads to duplication). This class only defines the interface. It should not dictate internal state of implementations.